test/fixtures/rules/std_rule_cases/L044.yml

```diff
@@ -514,3 +514,13 @@ test_snowflake_delete_cte:
   configs:
     core:
       dialect: snowflake
+
+test_pass_exasol_values_clause_cte:
+  pass_str: |
+    WITH txt AS (
+        VALUES (1)
+        AS t (id)
+    )
+
+    SELECT *
+    FROM txt
```

Comment:
This isn't valid snowflake/postgres/mysql
![image](https://user-images.githubusercontent.com/80432516/150544369-d86505b6-b378-437d-a942-164a8df90874.png)
so maybe use
```yml
configs:
    core:
      dialect: exasol
```
in these test cases

Reply:
This however is valid snowflake/postgres (although annoyingly doesn't parse in postgres atm) and also raises the same L044 issue:
- On main branch
![image](https://user-images.githubusercontent.com/80432516/150546227-0420e9f3-2867-40e4-8ecd-15c1f04344a9.png)

- On your branch:
![image](https://user-images.githubusercontent.com/80432516/150546677-e5cc37b5-8e32-4a41-bdf8-9befa2d53fbb.png)

L044 still there and catastophic failures 😱 🔥

Reply:
(I'll fix the postgres parsing for my example above since I know exactly where I've accidently introduced this, but that's not related to the above issue 😄 )

Reply:
Came up with the equivalent MySQL test case but it won't parse so I've added it into issue #2402. I have a feeling if it could parse then it would likely be solved by your fix for the postgres/snowflake example so we can leave it for a later date for now.

Reply:
Oops, I'll add the `exasol` config. I intended to do that, but forgot. Will also add the same test for `snowflake` dialect.

Reply:
I added a `postgres` test case as well.

Reply:
@jpy-git: Think we could create a separate issue for your `SELECT *` test case? I agree it should be fixed, but I think it's less likely to occur in production code. Happy to look into it, but I'd rather address it later (potentially next release).

Reply:
Hmm it's kind of the same as the existing test case just in a subquery rather than a CTE, and we'd be introducing catastrophic errors in rules L025 + L026. So don't think we should be merging this before at least the L025 + L026 [errors](https://github.com/sqlfluff/sqlfluff/pull/2400#discussion_r789718402) are fixed.

Worst case this runs into next release.

Reply:
I'll try and take a look this afternoon. I often have a couple of hours to work, then I don't always know when I'll have time again. 👶

Reply:
@jpy-git: I found a parsing issue with certain uses of `VALUES`. I created an issue in case you want to look at it: #2406. For now, I'll focus on the ones that parse correctly.